src/components/Faucet.tsx

```diff
@@ -0,0 +1,179 @@
+    const [withdrawalDelay, setWithdrawalDelay] = useState<bigint | null>(null);
+
+    const toastId = React.useRef<string | number | null>(null);
+
+    const checkEligibility = async () => {
+        toastId.current = toast.loading("Verifying Address...");
+        const response = await fetch('/api/faucet/verify', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ address: userAddress })
+        });
+        const result = await response.json();
+
+        if (response.ok) {
+            setVerification(true);
+            toast.update(toastId.current, {
+                render: result.message,
+                isLoading: false,
+                autoClose: 5000,
+                type: "success",
+            });
+        } else if (response.status === 400) {
+            setVerification(false);
+            setLastWithdrawalTime(BigInt(result.lastWithdrawalTime));
+            setWithdrawalDelay(BigInt(result.withdrawalDelay));
+            toast.update(toastId.current, {
+                render: result.message,
+                isLoading: false,
+                autoClose: 5000,
+                type: "error",
+            });
+        } else {
+            setVerification(false);
+            setLastWithdrawalTime(null);
+            setWithdrawalDelay(null);
+            toast.update(toastId.current, {
+                render: result.message || "Address is not verified 😢",
+                isLoading: false,
+                autoClose: 5000,
+                type: "error",
+            });
+        }
+    }
+
+    const sendFunds = async () => {
+        toastId.current = toast.loading("Sending Funds...");
+        const response = await fetch('/api/faucet/withdraw', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ address: userAddress })
+        });
+
+        const result = await response.json();
+
+        if (response.ok) {
+            const txHash = result.txHash;
+            setTxHash(txHash);
+            setVerification(false);
+            setLastWithdrawalTime(null);
+            setWithdrawalDelay(null);
+            toast.update(toastId.current, {
+                render: "Funds Sent Successfully! 🎉",
+                isLoading: false,
+                autoClose: 5000,
+                type: "success",
+            });
+        } else if (response.status === 500) {
+            toast.update(toastId.current, {
+                render: result.message || "Failed to send funds 😢",
+                isLoading: false,
+                autoClose: 5000,
+                type: "error",
+            });
+        }
+    }
+
+    const formatDate = (timestamp: bigint | string) => {
+        let numericTimestamp: number;
+
+        if (typeof timestamp === 'string') {
```

Comment:
you can use ternanry operator

since u have defined timestamp's types in the function parameters, you don't need to add a case for it not being either of them

you can do:

```
numericTimestamp =  typeof timestamp === 'string' ? parseInt(timestamp, 16):  Number(timestamp;

```